torch/csrc/jit/tensorexpr/expr.cpp

```diff
@@ -50,6 +50,20 @@ ExprHandle ExprHandle::operator<=(const ExprHandle& other) const {
   return CompareSelect::make(*this, other, CompareSelectOperation::kLE);
 }
 
+ExprHandle ExprHandle::operator&&(const ExprHandle& other) const {
+  if (!this->node()->dtype().is_integral()) {
+    throw unsupported_dtype();
+  }
+  return IfThenElse::make(*this, other, IntImm::make(0));
+}
+
+ExprHandle ExprHandle::operator||(const ExprHandle& other) const {
+  if (!this->node()->dtype().is_integral()) {
+    throw unsupported_dtype();
+  }
+  return IfThenElse::make(*this, IntImm::make(1), other);
```

Comment:
The immediate must have the same dtype as `other`, otherwise IR will be malformed. For instance, if `other` is `bool`, then we need to use `BoolImm::make(1)` instead of `IntImm::make(1)`. Also I think it would be worth adding a test for exercising different dtypes as well.